app/src/main/java/net/skyscanner/backpack/demo/compose/AppSearchModalStory.kt

```diff
@@ -0,0 +1,232 @@
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.unit.dp
+import net.skyscanner.backpack.compose.appsearchmodal.BpkAction
+import net.skyscanner.backpack.compose.appsearchmodal.BpkAppSearchModal
+import net.skyscanner.backpack.compose.appsearchmodal.BpkAppSearchModalResult
+import net.skyscanner.backpack.compose.appsearchmodal.BpkItem
+import net.skyscanner.backpack.compose.appsearchmodal.BpkSection
+import net.skyscanner.backpack.compose.appsearchmodal.BpkSectionHeading
+import net.skyscanner.backpack.compose.appsearchmodal.BpkShortcut
+import net.skyscanner.backpack.compose.button.BpkButton
+import net.skyscanner.backpack.compose.icon.BpkIcon
+import net.skyscanner.backpack.compose.tokens.Airports
+import net.skyscanner.backpack.compose.tokens.City
+import net.skyscanner.backpack.compose.tokens.Landmark
+import net.skyscanner.backpack.compose.tokens.UseLocation
+import net.skyscanner.backpack.demo.R
+import net.skyscanner.backpack.demo.components.AppSearchModalComponent
+import net.skyscanner.backpack.demo.meta.ComposeStory
+
+@Composable
+@AppSearchModalComponent
+@ComposeStory("Content")
+fun AppSearchModalStoryContent(modifier: Modifier = Modifier) {
+    AppSearchModalStory(result = contentResult(LocalContext.current.resources))
+}
+
+@Composable
+@AppSearchModalComponent
+@ComposeStory("Loading")
+fun AppSearchModalStoryLoading(modifier: Modifier = Modifier) {
+    AppSearchModalStory(result = loadingResult(LocalContext.current.resources))
+}
+
+@Composable
+@AppSearchModalComponent
+@ComposeStory("Error")
+fun AppSearchModalStoryError(modifier: Modifier = Modifier) {
+    AppSearchModalStory(result = errorResult(LocalContext.current.resources))
+}
+
+@Composable
+private fun AppSearchModalStory(
+    result: BpkAppSearchModalResult,
+    modifier: Modifier = Modifier,
+) {
+    Column(modifier.fillMaxSize()) {
+        DefaultAppSearchModalSample(result = result)
+    }
+}
+
+@Composable
+internal fun DefaultAppSearchModalSample(
+    result: BpkAppSearchModalResult,
+    modifier: Modifier = Modifier,
+) {
+    val showModal = rememberSaveable { mutableStateOf(true) }
+    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+        BpkButton(
+            text = stringResource(R.string.generic_show),
+            onClick = { showModal.value = true },
+        )
+    }
+    if (showModal.value) {
+        BpkAppSearchModal(
+            modifier = modifier,
+            title = stringResource(id = R.string.destination),
+            inputText = "",
```

Comment:
I didn't want to because we already using `BpkTextField` which should have its own tests already

Reply:
you don't trust me 😞, just kidding, its a good point, that could happen, I will add test to make sure it will be highlighted in case someone changes it accidentally